development.md

```diff
@@ -1,152 +1,109 @@
+## Prerequisites
 
-# Set up Git & GitHub
+To build and run the app from source, you'll need the following:
 
-You can skip this step if you already have Git, GitHub.
+* [Git](http://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+  * Use our [Git Guide](https://zulip.readthedocs.io/en/latest/git/setup.html) to get started with Git and GitHub.
+* [Node.js](https://nodejs.org) >= v6.9.0
+  * [NPM](https://www.npmjs.com/get-npm) and
+    [node-gyp](https://github.com/nodejs/node-gyp#installation),
+    if they don't come bundled with your Node.js installation
+* [Python](https://www.python.org/downloads/release/python-2713/)
+  (v2.7.x recommended)
+* A C++ compiler compatible with C++11
+* Linux users also need [Snapcraft](https://snapcraft.io/)
+* Development headers for the libXext, libXtst, and libxkbfile libraries
 
-Follow our [Git Guide](https://zulip.readthedocs.io/en/latest/git/setup.html) in order to install Git, set up a GitHub account
+### Ubuntu/Linux and other Debian-based distributions
 
+On a system running Debian, Ubuntu, or another Debian-based Linux
+distribution, you can install all dependencies through the package
+manager (see [here][node-debian] for more on the first command):
 
-# Install Prerequisites
+```sh
+$ curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
+$ sudo apt install git nodejs python build-essential snapcraft libxext-dev libxtst-dev libxkbfile-dev libgconf-2-4
+```
 
-Jump to:
+[node-debian]: https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions
 
-- [MacOS](https://github.com/zulip/zulip-electron/blob/master/development.md#macos)
-- [Ubuntu/Debian](https://github.com/zulip/zulip-electron/blob/master/development.md#ubuntudebian)
-- [Windows](https://github.com/zulip/zulip-electron/blob/master/development.md#windows)
+### MacOS
 
+On a system running MacOS, you can refer to [official nodejs docs][node-mac] to
+install nodejs. To ensure Node.js has been installed, run  ```node -v``` in terminal to know your node version.
 
-## MacOS
+[node-mac]: https://nodejs.org/en/download/package-manager/#macos
 
+If [NPM](https://www.npmjs.com/get-npm) and [node-gyp](https://github.com/nodejs/node-gyp#installation) don't come bundled with your Node.js installation, you will need to install them manually. 
 
-  **Node JS**
-  Go to the [Node.js Downloads page](https://nodejs.org/en/download/). Download Node.js for MacOS (`v6.9.0` or above recommended). Run the downloaded Node.js `.pkg` Installer. You're finished! To ensure Node.js has been installed, run `node -v` in your terminal - you should get something like `v6.9.0` or above 
+### Windows
```

Comment:
On Windows, I think we need to install some global dependencies as well. Probably - `npm install --global --production windows-build-tools` as admin.